### PR TITLE
SONARPY-1900: Extend S6973 - Important hyperparameters should be specified for Scikit-Learn estimators

### PR DESCRIPTION
**Pull request by Seppli11**

### CODE REVIEW

File: python-checks/src/test/resources/checks/pytorch_estimator_hyperparameters.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+def sgd(param_dict, param_list, dampening):
+    from torch.optim import SGD
+    noncompliant = SGD() # Noncompliant
+    noncompliant = SGD(model.parameters(), lr = 0.001) # Noncompliant
+    noncompliant = SGD(model.parameters(), weight_decay = 0.001) # Noncompliant
+    noncompliant = SGD(model.parameters(), momentum = 0.001) # Noncompliant
+
+    optimizer = SGD(model.parameters(), 0.001, 0.323, dampening, 0.23)
+    optimizer = SGD(model.parameters(), lr = 0.001, momentum=0.323, weight_decay=0.23)
+    optimizer = SGD(model.parameters(), **param_dict, *param_list)
```
Comment (joke1196) [inline python-checks/src/test/resources/checks/pytorch_estimator_hyperparameters.py:156]: I am not sure why but the last two lines of each function are duplicated. Also I would suggest not to test multiple times the same thing.  For example on line 156 your are testing that we do not raise on spread operator, but this is already tested on line 145.

**Comment by joke1196** [comment]
Usually we use the name of the ticket as the title of the PR, this is FYI, it is not really important as long as the commit message when squashed is correct.

**Review by joke1196 — CHANGES_REQUESTED** [review]
This looks quite good. Thank you! I have added a few comments and suggestions. 
I appreciate you split the work into multiple commit, the only thing I could suggest doing here is to do the refactoring (from one class to a private nested class) for SkLearn first in its dedicated commit, 
and in a second time add the PyTorch implementation. This would allow the reviewer to not have a file completely new to review.
This would also help in making sure nothing was broken during the refactoring right away.

File: python-checks/src/main/java/org/sonar/python/checks/SklearnPyTorchEstimatorHyperparametersCheck.java [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+      Map.entry("torch.optim.SGD", List.of(new Param(LR, 1), new Param("momentum", 2), new Param(WEIGHT_DECAY, 4)))
+    );
+
+    public static Optional<List<Param>> getMissingParameters(String name, CallExpression callExpression) {
+      return Optional.ofNullable(PY_TORCH_ESTIMATORS_AND_PARAMETERS_TO_CHECK.get(name))
+        .filter(parameters -> !containsSpreadOperator(callExpression))
+        .filter(parameters -> isMissingAHyperparameter(callExpression, parameters));
+    }
+
+    private static boolean containsSpreadOperator(CallExpression callExpression) {
```
[the remark this answers is not in the document]
Comment (Seppli11) [reply python-checks/src/main/java/org/sonar/python/checks/SklearnPyTorchEstimatorHyperparametersCheck.java:136]: Would it make sens to do this in a separate PR?

[the remark this answers is not in the document]
Comment (Seppli11) [reply python-checks/src/main/java/org/sonar/python/checks/SklearnPyTorchEstimatorHyperparametersCheck.java]: @joke1196 mentioned that this should be done in a new PR. The issue is https://sonarsource.atlassian.net/browse/SONARPY-2140

**Review by joke1196 — APPROVED** [review]
LGTM!

**Pull request merged**